library/src/main/java/com/novoda/downloadmanager/NotificationDispatcher.java

```diff
@@ -0,0 +1,45 @@
+package com.novoda.downloadmanager;
+
+import static com.novoda.downloadmanager.DownloadBatchStatus.Status.DELETION;
+import static com.novoda.downloadmanager.DownloadBatchStatus.Status.DOWNLOADED;
+
+class NotificationDispatcher {
+
+    private final Object waitForDownloadService;
+    private final NotificationCreator<DownloadBatchStatus> notificationCreator;
+
+    private DownloadService downloadService;
+
+    NotificationDispatcher(Object waitForDownloadService, NotificationCreator<DownloadBatchStatus> notificationCreator) {
+        this.waitForDownloadService = waitForDownloadService;
+        this.notificationCreator = notificationCreator;
+    }
+
+    void updateNotification(DownloadBatchStatus downloadBatchStatus) {
+        WaitForDownloadService.<Void>waitFor(downloadService, waitForDownloadService)
+                .thenPerform(executeUpdateNotification(downloadBatchStatus));
+    }
+
+    private WaitForDownloadService.ThenPerform.Action<Void> executeUpdateNotification(DownloadBatchStatus downloadBatchStatus) {
+        return () -> {
+            NotificationInformation notificationInformation = notificationCreator.createNotification(downloadBatchStatus);
+
+            if (downloadBatchStatus.status() == DELETION) {
```

Comment:
How about : 
```
if () {
} else if (){
} else {
}
return null;
```

Reply:
I prefer the quickness and readability of the `if -> return` 😄  I guess its a preference thing!

Reply:
What I don't like in this scenario that `else` is not really highlighted enough. But don't mind really.

Reply:
6ffb008